framework/midi/internal/zerberussynth.cpp

```diff
@@ -72,27 +72,24 @@ Ret ZerberusSynth::addSoundFont(const io::path& filePath)
     return true;
 }
 
-Ret ZerberusSynth::setupChannels(const Programs& programs)
+bool ZerberusSynth::setupChannels(const std::vector<Event>& events)
 {
     IF_ASSERT_FAILED(m_zerb) {
         return false;
     }
 
-    m_programs = programs;
-
-    for (const Program& prog : m_programs) {
-        m_zerb->controller(prog.channel, Ms::CTRL_PROGRAM, prog.program);
+    for (const Event& e : events) {
+        handleEvent(e);
     }
 
     return make_ret(Ret::Code::Ok);
```

Comment:
make_ret return an instance of Ret, but the function has return type bool

Reply:
fixed